client/payments-welcome/index.tsx

```diff
@@ -0,0 +1,288 @@
+
+	const renderErrorMessage = ( message: string ) => {
+		setErrorMessage( message );
+		setSubmitted( false );
+	};
+
+	const activatePromo = async () => {
+		try {
+			// const activatePromoRequest = ( await apiFetch( {
+			// 	path: '/wc-calypso-bridge/v1/payments/activate-promo',
+			// 	method: 'POST',
+			// } ) ) as activatePromoResponse;
+
+			// if ( activatePromoRequest?.success ) {
+			window.location.href = connectUrl;
+			// }
+		} catch ( e ) {
+			// renderErrorMessage( e.message );
+		}
+	};
+
+	const handleSetup = async () => {
+		setSubmitted( true );
+		recordEvent( 'wcpay_connect_account_clicked', {
+			// eslint-disable-next-line camelcase
+			wpcom_connection: isJetpackConnected ? 'Yes' : 'No',
+		} );
+
+		try {
+			const installAndActivateResponse = await installAndActivatePlugins(
+				[ 'woocommerce-payments' ]
+			);
+
+			if ( installAndActivateResponse?.success ) {
+				activatePromo();
+			} else {
+				renderErrorMessage( installAndActivateResponse.message );
+			}
+		} catch ( e ) {
+			renderErrorMessage( e.message );
+		}
+	};
+
+	const handleNoThanks = () => {
+		setNoThanksClicked( true );
+		setExitSurveyModalOpen( true );
+	};
+
+	return (
+		<Card className="connect-account__card">
+			<CardHeader>
+				<div>
+					<h1 className="banner-heading-copy">
+						{ strings.bannerHeading }
+					</h1>
+					<TermsOfService />
+				</div>
+				<div className="connect-account__action">
+					<Button
+						isSecondary
+						isBusy={ isNoThanksClicked && isExitSurveyModalOpen }
+						disabled={ isNoThanksClicked && isExitSurveyModalOpen }
+						onClick={ handleNoThanks }
+						className="btn-nothanks"
+					>
+						{ strings.nothanks }
+					</Button>
+					<Button
+						isPrimary
+						isBusy={ isSubmitted }
+						disabled={ isSubmitted }
+						onClick={ handleSetup }
+						className="btn-install"
+					>
+						{ strings.button }
+					</Button>
+					{ isExitSurveyModalOpen && (
+						<ExitSurveyModal
+							setExitSurveyModalOpen={ setExitSurveyModalOpen }
+						/>
+					) }
+				</div>
+			</CardHeader>
+			<CardBody>
+				<div className="content">
+					<p className="onboarding-description">
+						{ strings.onboarding.description }
+						<br />
+						<LearnMore />
+					</p>
+
+					<h3 className="accepted-payment-methods">
+						{ strings.paymentMethodsHeading }
+					</h3>
+
+					<PaymentMethods />
+				</div>
+			</CardBody>
+		</Card>
+	);
+};
+
+const ConnectAccountPage = () => {
+	const [ errorMessage, setErrorMessage ] = useState( '' );
+
+	const { isJetpackConnected, connectUrl, hasViewedWelcomePage } = useSelect(
+		( select ) => {
+			const { getOption } = select( OPTIONS_STORE_NAME );
+			return {
+				isJetpackConnected: select(
+					'wc/admin/plugins'
+				).isJetpackConnected(),
+				connectUrl:
+					'admin.php?wcpay-connect=1&_wpnonce=' +
+					getSetting( 'wcpay_welcome_page_connect_nonce' ),
+				hasViewedWelcomePage:
```

Comment:
Seems like the type of `hasViewedWelcomePage ` can be string or boolean. The hasViewedWelcomePage` string might be a little strange. How about making this to be a boolean value only?

Reply:
I totally agree. The reason it can be string or boolean is that WP stores all options in string. The `getOption` function also returns `undefined` when the request is not finished. I actually renamed the variable to `pageViewTimestamp` and changed the if statement condition so it makes more sense.

Please check https://github.com/woocommerce/woocommerce-admin/pull/8083/commits/05a5974d078c5879d4588abe25f6c352ce46e2e3